core/src/main/java/ar/com/wolox/wolmo/core/permission/PermissionManager.kt

```diff
@@ -0,0 +1,121 @@
+     * was already granted, **false** otherwise.
+     */
+    fun requestPermission(fragment: Fragment, listener: PermissionListener?, vararg permissions: String): Boolean {
+        val ungrantedPermissions = filterUngranted(*permissions)
+        if (ungrantedPermissions.isNotEmpty()) {
+            fragment.requestPermissions(ungrantedPermissions, requestCount)
+            requestListeners.put(requestCount++, listener)
+            return false
+        }
+        listener?.onPermissionsGranted()
+        return true
+    }
+
+    /**
+     * Request one or more Android's runtime permissions using an [activity] to perform the
+     * request, [listener] to receive the response of the request (can be null if not
+     * interested in the response) and [permissions] ( One or more [String] objects with
+     * the permissions to be requested. Returns **true** if every requested permission
+     * was already granted, **false** otherwise.
+     */
+    fun requestPermission(activity: Activity, listener: PermissionListener?, vararg permissions: String): Boolean {
+        val ungrantedPermissions = filterUngranted(*permissions)
+        if (ungrantedPermissions.isNotEmpty()) {
+            ActivityCompat.requestPermissions(activity, ungrantedPermissions, requestCount)
+            requestListeners.put(requestCount++, listener)
+            return false
+        }
+        listener?.onPermissionsGranted()
+        return true
+    }
+
+    /**
+     * Callback that must be called from Activities and Fragments that will make use of
+     * [PermissionManager]. The [requestCode] is used to get the requestListeners. If
+     * listener is not null, removes [requestCode] from the requestListeners.
+     * The requested [permissions] are never null. The [grantResults] for the
+     * corresponding permissions is either
+     * [android.content.pm.PackageManager.PERMISSION_GRANTED] or
+     * [android.content.pm.PackageManager.PERMISSION_DENIED]. Never null.
+     */
+    fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+        requestListeners.get(requestCode)?.let {
+            requestListeners.remove(requestCode)
+            // Workaround to Android bug: https://goo.gl/OwseuO
+            Handler().post {
+                if (allGranted(grantResults)) {
+                    it.onPermissionsGranted()
+                } else {
+                    it.onPermissionsDenied(filterUngranted(*permissions))
+                }
+            }
+        }
+    }
+
+    /** Filters a list of [permissions] and returns only the ones which have not been granted. */
+    private fun filterUngranted(vararg permissions: String): Array<String> {
+        val ungranted = ArrayList<String>()
+        permissions.filter { permission -> ContextCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED }.forEach { ungranted.add(it) }
+        return ungranted.toTypedArray()
+    }
```

Comment:
Use `.toTypedArray()` after the `filter` instead of the foreach and the new list.

If you want: 👀 
```
fun filterUngranted(vararg permissions: String) = permissions.filter { permission ->
        ContextCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED
}.toTypedArray()
```

Reply:
I like this